crates/bevy_tasks/src/task_pool.rs

```diff
@@ -216,71 +139,11 @@ impl TaskPool {
-        // transmute the lifetimes to 'env here to appease the compiler as it is unable to validate safety.
-        let executor: &async_executor::Executor = &*self.executor;
-        let executor: &'env async_executor::Executor = unsafe { mem::transmute(executor) };
-        let task_scope_executor = &async_executor::Executor::default();
-        let task_scope_executor: &'env async_executor::Executor =
-            unsafe { mem::transmute(task_scope_executor) };
-        let spawned: ConcurrentQueue<async_executor::Task<T>> = ConcurrentQueue::unbounded();
-        let spawned_ref: &'env ConcurrentQueue<async_executor::Task<T>> =
-            unsafe { mem::transmute(&spawned) };
-
-        let scope = Scope {
-            executor,
-            task_scope_executor,
-            spawned: spawned_ref,
-            scope: PhantomData,
-            env: PhantomData,
-        };
-
-        let scope_ref: &'env Scope<'_, 'env, T> = unsafe { mem::transmute(&scope) };
-
-        f(scope_ref);
-
-        if spawned.is_empty() {
-            Vec::new()
-        } else {
-            let get_results = async move {
-                let mut results = Vec::with_capacity(spawned.len());
-                while let Ok(task) = spawned.pop() {
-                    results.push(task.await);
-                }
-
-                results
-            };
-
-            // Pin the futures on the stack.
-            pin!(get_results);
-
-            // SAFETY: This function blocks until all futures complete, so we do not read/write
-            // the data from futures outside of the 'scope lifetime. However,
-            // rust has no way of knowing this so we must convert to 'static
-            // here to appease the compiler as it is unable to validate safety.
-            let get_results: Pin<&mut (dyn Future<Output = Vec<T>> + 'static + Send)> = get_results;
-            let get_results: Pin<&'static mut (dyn Future<Output = Vec<T>> + 'static + Send)> =
-                unsafe { mem::transmute(get_results) };
-
-            // The thread that calls scope() will participate in driving tasks in the pool
-            // forward until the tasks that are spawned by this scope() call
-            // complete. (If the caller of scope() happens to be a thread in
-            // this thread pool, and we only have one thread in the pool, then
-            // simply calling future::block_on(spawned) would deadlock.)
-            let mut spawned = task_scope_executor.spawn(get_results);
-
-            loop {
-                if let Some(result) = future::block_on(future::poll_once(&mut spawned)) {
-                    break result;
-                };
-
-                self.executor.try_tick();
-                task_scope_executor.try_tick();
-            }
-        }
+        self.0.scope(|s| {
+            let scope = Scope(s);
+            let scope_ref: &'env Scope<'_, 'env, T> = unsafe { mem::transmute(&scope) };
```

Comment:
I'm not an expert with unsafe Rust, but `'_` probably creates an [unbounded lifetime](https://doc.rust-lang.org/nomicon/transmutes.html) (See the unbounded lifetimes part). You should specify a lifetime for `'scope` in `Scope`.

Reply:
Tested with `<'scope, 'env, T>`, same SIGSEGV

Reply:
I posted this on discord, but will repost here so it doesn't get lost. The problem is that scope_ref only lives for the lifetime of the closure it's in, but needs to live until `.scope` returns. I'm not sure how to get around this as it's important that the scope object is constructed inside the scope function, so that the `'scope` lifetime is only available inside the scope function. Basically this needs to be `self.0.scope(f)`, but I'm not sure how you do that with the wrappers.